File: app/src/main/java/com/playgilround/schedule/client/main/MainActivity.java
package com.playgilround.schedule.client.main;

import android.os.Bundle;
import com.google.android.material.tabs.TabLayout;
import com.playgilround.schedule.client.R;
import com.playgilround.schedule.client.calendarschedule.CalendarScheduleFragment;
import com.playgilround.schedule.client.chat.ChatFragment;
import com.playgilround.schedule.client.friend.FriendFragment;
import com.playgilround.schedule.client.main.view.MainViewPagerAdapter;
import com.playgilround.schedule.client.map.MapFragment;
import com.playgilround.schedule.client.setting.SettingFragment;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;
import androidx.viewpager.widget.ViewPager;
import butterknife.BindView;
import butterknife.ButterKnife;

public class MainActivity extends AppCompatActivity implements MainContract.View {

    @BindView(R.id.view_pager_main)
    ViewPager mViewPager;

    @BindView(R.id.nav_main)
    TabLayout mTabLayout;

    private MainContract.Presenter mPresenter;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        ButterKnife.bind(this);
        new MainPresenter(this);

        MainViewPagerAdapter pagerAdapter = new MainViewPagerAdapter(getSupportFragmentManager());

        pagerAdapter.addFragment(R.drawable.image_nav_bar_friend, new FriendFragment());
        pagerAdapter.addFragment(R.drawable.image_nav_bar_chat, new ChatFragment());
        pagerAdapter.addFragment(R.drawable.image_nav_bar_calendar, new CalendarScheduleFragment());
        pagerAdapter.addFragment(R.drawable.image_nav_bar_map, new MapFragment());
        pagerAdapter.addFragment(R.drawable.image_nav_bar_setting, new SettingFragment());
        mViewPager.setAdapter(pagerAdapter);

        mTabLayout.setupWithViewPager(mViewPager);

        for (int i =0; i< mViewPager.getAdapter().getCount(); i++) {
            mTabLayout.getTabAt(i).setIcon(pagerAdapter.getMImageList().get(i));
        }
    }
    @Override
    public void setPresenter(MainContract.Presenter presenter) {
        mPresenter = presenter;
    }
}

File: app/src/main/java/com/playgilround/schedule/client/main/MainActivity.kt
package com.playgilround.schedule.client.main

import android.os.Bundle
import android.os.PersistableBundle
import androidx.appcompat.app.AppCompatActivity
import com.playgilround.schedule.client.R
import com.playgilround.schedule.client.calendarschedule.CalendarScheduleFragment
import com.playgilround.schedule.client.chat.ChatFragment
import com.playgilround.schedule.client.friend.FriendFragment
import com.playgilround.schedule.client.main.view.MainViewPagerAdapter
import com.playgilround.schedule.client.map.MapFragment
import com.playgilround.schedule.client.setting.SettingFragment
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity: AppCompatActivity(), MainContract.View {

    private lateinit var mPresenter: MainContract.Presenter

    private val drawableList = arrayOf(
            R.drawable.image_nav_bar_friend,
            R.drawable.image_nav_bar_chat,
            R.drawable.image_nav_bar_calendar,
            R.drawable.image_nav_bar_map,
            R.drawable.image_nav_bar_setting
    )

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        MainPresenter(this)

        val pagerAdapter = MainViewPagerAdapter(supportFragmentManager)
        pagerAdapter.addFragment(drawableList[0], FriendFragment())
        pagerAdapter.addFragment(drawableList[1], ChatFragment())
        pagerAdapter.addFragment(drawableList[2], CalendarScheduleFragment())
        pagerAdapter.addFragment(drawableList[3], MapFragment())
        pagerAdapter.addFragment(drawableList[4], SettingFragment())

        view_pager_main.adapter = pagerAdapter

        nav_main.setupWithViewPager(view_pager_main)

        for (i in 0 until view_pager_main.adapter!!.count) {
            nav_main.getTabAt(i)!!.setIcon(pagerAdapter.mImageList[i])
        }

    }

    override fun setPresenter(presenter: MainContract.Presenter) {
        mPresenter = presenter
    }
}